src/Bonnier/WP/Cache/Models/Post.php

```diff
@@ -8,37 +8,106 @@
+        add_action('publish_to_draft', [__CLASS__, 'remove_post'], 10, 1);
+
+        // publish post
+        add_action('publish_to_publish', [__CLASS__, 'update_posts'], 10, 1);
+
+        add_action('draft_to_publish', [__CLASS__, 'publish_post'], 10, 1);
+        add_action('untrashed_post', [__CLASS__, 'publish_post'], 10, 1);
+    }
+
+    public static function publish_post($publishedPostID)
+    {
+        $publishedPost = get_post($publishedPostID);
+
+        // Skip update on attachment & inherit
+        if (! ('publish' === $publishedPost->post_status
+                || ('attachment' === get_post_type($publishedPost) && 'inherit' === $publishedPost->post_status))
+            || is_post_type_hierarchical($publishedPost->post_type)
+        ) {
+            return;
+        }
+
+        CacheApi::add($publishedPost->ID);
     }
 
-    public static function post_status_changed($new_status, $old_status, $post)
+    public static function update_posts($changedPostID)
     {
-        //Ignore deleted draft posts
-        if ($old_status === 'draft' && $new_status === 'trash') {
+        $changedPost = get_post($changedPostID);
+
+        global $post;
+        $deleteOldFlag = false;
+
+        if (!('publish' === $changedPost->post_status
+                || ('attachment' === get_post_type($changedPost)
+                    && 'inherit' === $changedPost->post_status)
+            ) || is_post_type_hierarchical($changedPost->post_type)
+        ) {
             return;
         }
 
-        if ($new_status === 'publish') {
-            self::update_post($post->ID);
-            //If post is trashed or drafted
-        } elseif ($new_status === 'trash' || $new_status === 'draft') {
-            self::delete_post($post->ID);
+        // Post name has changed. Clean old URL
+        if ($changedPost->post_name != $post->post_name) {
+            $deleteOldFlag = true;
         }
+
+        // Check if the category has changed. If so, clean old URL
+        $newPostCategory = get_term($_REQUEST['acf'][static::ACF_CATEGORY_ID]);
+        $oldPostCategory = get_the_category($post->ID)[0];
+        if ($newPostCategory->term_id != $oldPostCategory->term_id) {
+            $deleteOldFlag = true;
+        }
+
+        // Delete if flagged
+        if ($deleteOldFlag) {
+            CacheApi::delete($changedPost->ID);
+
+            // If delete is triggered add(!) the new one.
+            CacheApi::add($changedPost->ID);
+            return;
+        }
+
+        CacheApi::update($changedPost->ID);
+    }
+
+    public static function remove_post($postID)
+    {
+
+        // if the post we try to trash is current published,
+
+        CacheApi::delete($postID);
     }
 
+    /**
+     * @param $postId
+     * @return bool
+     * @deprecated
+     */
     public static function update_post($postId)
     {
         return CacheApi::update($postId);
     }
 
+    /**
+     * @param $postId
+     * @return bool|void
+     * @deprecated
```

Comment:
why we have @deprecated inside the phpdoc ?

Reply:
I would like to get them removed later on, however I'm not sure if any of our others plugins are using them. it should be followed up on, so that's why I added the `@deprecated` tag